### PR TITLE
[INFRA-2908] Prepare to publish Stapler via JEP-229

### PR DESCRIPTION
**Pull request by jglick**
See https://issues.jenkins.io/browse/INFRA-2908 for discussion. Downstream of https://github.com/jenkins-infra/repository-permissions-updater/pull/1950 and https://github.com/jenkins-infra/javadoc/pull/29.

### CODE REVIEW

File: docs/reference.adoc [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+-> evaluate(, "/docsAndFiles/upload")
+-> evaluate(.getDocsAndFiles(), "/upload")
+-> evaluate(, "/upload")
+-> .doUpload(...)
+....
+____
+
+The exact manner of recursion depends on the signature of the type of
+the `node` parameter, and the following sections describe them in
+detail. Also see
```
Comment (timja) [inline docs/reference.adoc:34]: dead document link, can be fixed up later though

File: docs/reference.adoc [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+variables represent a single token, while upper-case variables represent
+variable-length tokens. For example, if we say `url=[a,W]` and the
+actual URL was "/abc/def/ghi", then `a="abc"` and `W="/def/ghi"`. If the
+actual URL was "/abc", then `a="abc"` and `W=""`.
+____
+
+==== Stapler Proxy
+
+`node` can implement
+https://stapler.dev.java.net/nonav/javadoc/index.html?org/kohsuke/stapler/StaplerProxy.html[the
```
Comment (timja) [inline docs/reference.adoc:64]: dead

File: docs/what-is.adoc [hunk 3]
```diff
@@ -0,0 +1,88 @@
+Stapler is a library that "staples" your application objects to URLs,
```
Comment (timja) [inline docs/what-is.adoc:1]:
```suggestion
= What is Stapler?

Stapler is a library that "staples" your application objects to URLs,
```

**Review by timja — APPROVED** [review]
🚢 ?

**Comment by jglick** [comment]
Uninstalled Release Drafter App. Going to see if this works…

**Comment by jglick** [comment]
https://repo.jenkins-ci.org/releases/org/kohsuke/stapler/stapler/1527.ve41b3ce15c05/ https://github.com/stapler/stapler/runs/2522813401?check_suite_focus=true https://github.com/stapler/stapler/releases/tag/1527.ve41b3ce15c05 :rocket:

**Pull request merged**